### PR TITLE
Add-DbaAgDatabase - Add missing SqlCredential when calling Test-DbaAvailabilityGroup (part 2)

### LINKED ISSUE #7481: [Bug]Add-DbaAgDatabase does not honor SqlCredential
Add-DbaAgDatabase was recently re-written and the re-write does not honor the passed in SqlCredential parameter.  
https://github.com/sqlcollaborative/dbatools/commit/c864b791ad360d954a93edc065c691e117714a2e

When Add-DbaAgDatabase calls Test-DbaAvailabilityGroup, it should pass along the SqlCredential parameter if one was passed into Add-DbaAgDatabase at runtime.

This change breaks automation scenarios where dba tools is running under an account that does not have access to SQL.

### PR DESCRIPTION
**Pull request by andreasjordan**
## Type of Change

 - [x] Bug fix (non-breaking change, fixes #7481  )
 - [ ] New feature (non-breaking change, adds functionality, fixes # )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1`)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/dataplat/appveyor-lab ?
 - [ ] Unit test is included
 - [ ] Documentation
 - [ ] Build system

See also #7482.

### CODE REVIEW

File: functions/Add-DbaAgDatabase.ps1 [hunk 1]
```diff
@@ -211,6 +211,7 @@ function Add-DbaAgDatabase {
                 Write-Progress @progress
                 $testSplat = @{
                     SqlInstance            = $db.Parent
+                    SqlCredential          = $SqlCredential
```
Comment (potatoqualitee) [inline functions/Add-DbaAgDatabase.ps1:214]: Thanks Andreas! $db.Parent is a connected object and shouldn't need SqlCredential. Does this really fix the issue?

Comment (potatoqualitee) [reply functions/Add-DbaAgDatabase.ps1:214]: Perhaps a .Refresh() is needed somewhere instead?

Comment (andreasjordan) [reply functions/Add-DbaAgDatabase.ps1:214]: Thinking again, you are probably right. Could be the reason why I have not added that in the first round....

Comment (potatoqualitee) [reply functions/Add-DbaAgDatabase.ps1:214]: Sometimes things are weird. If you want, you can use $db.Parent.Name and $SqlCredential to force a new connect or a refresh may help.

**Pull request closed without merging**